src/main.ts

```diff
@@ -19,22 +19,31 @@ export async function main(): Promise<boolean> {
     // Get all installed packages
     let packages = await getInstalledPackages(configuration);
     if (packages.length === 0) {
-        return false;
+        // No dependency -> no compliance issue
+        return true;
     }
 
     // Filter packages
     packages = excludePackages(packages, configuration);
 
     const report = FactoryReport.getInstance(configuration.report, configuration.format);
 
-    // Verify allowed licenses
-    const invalidPackages = onlyAllow(packages, configuration);
-    if (invalidPackages.length > 0) {
-        report.process(invalidPackages);
-        return false;
+    // Verify allowed licenses: command behaviour will be different whether "allow" is set or not
+    const isComplianceModeEnabled = Array.isArray(configuration.allow) && configuration.allow.length > 0;
```

Comment:
Create a method in the configuration module to remove responsibility from main.
```javascript
if (isComplianceEnabled()) {
```

Reply:
Done, I have added basic unit tests for it as well